app/models/grda_warehouse/tasks/task_instrumentation.rb

```diff
@@ -9,7 +10,18 @@ def self.call(...) = instance.call(...)
   def call(name, alert_threshold:, &block)
     task = find_or_create_maintenance_task(name, alert_threshold: alert_threshold)
     run = task.system_maintenance_task_runs.create!(started_at: Time.current)
-    block.call(run)
+    if Rails.env.production?
+      block.call(run)
+    else
+      report = MemoryProfiler.report do
+        block.call(run)
+      end
+      run.update!(
+        memory_allocated_mb: report.total_allocated_memsize / 1024 / 1024,
+        memory_retained_mb: report.total_retained_memsize / 1024 / 1024,
```

Comment:
this is fine as is. I might just store the bytes and leave the formatting to number_to_human_size but not worth the trouble